sparse-array/src/fast.rs

```diff
@@ -0,0 +1,187 @@
+///! Wrapper for sparse arrays that doesn't delete anything from the memory
+/// unless `shrink` is called.
+
+use std::sync::Arc;
+use crate::{
+    bitmap::BitmapIndex,
+    SparseArray,
+};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct FastSparseArray<V> {
+    index: BitmapIndex,
+    data: Arc<SparseArray<V>>,
+}
+
+impl<V: Clone> FastSparseArray<V> {
```

Comment:
I don't think `Clone on V` should be needed for every method. I can only imagine `remove` and `shrink` need it since it's immutable, but that's about it.